src/include/sof/sof.h

```diff
@@ -28,6 +39,40 @@ struct sof {
 
 	/* DMA for Trace*/
 	struct dma_trace_data *dmat;
-};
+
+	/* generic trace structure */
+	struct trace *trace;
+
+	/* platform clock information */
+	struct clock_info *clocks;
+
+	/* default platform timer */
+	struct timer *platform_timer;
+
+	/* timer domain for driving timer LL scheduler */
+	struct ll_schedule_domain *platform_timer_domain;
+
+	/* DMA domain for driving DMA LL scheduler */
+	struct ll_schedule_domain *platform_dma_domain;
+
+	/* memory map */
+	struct mm *memory_map;
+
+	/* runtime power management data */
+	struct pm_runtime_data *prd;
+
+	/* shared notifier data */
+	struct notify_data *notify_data;
+
+	/* platform dai information */
+	struct dai_info *dai_info;
+
+	/* platform DMA information */
+	struct dma_info *dma_info;
+
+	__aligned(PLATFORM_DCACHE_ALIGN) int alignment[0];
```

Comment:
Why is this needed? Do we want to protect some objects that may come in memory after 'sof' object, so they will not get partially de/cached?

Reply:
@jajanusz Exactly. To be 100% sure that slaves core won't mess up the the rest of the data for master core we need to align ending of the sof object to cache line. Currently before slave core enablement we writeback the whole bss.

Reply:
It's probably worth adding some comments next to the main sof structure explaining this.

Reply:
@lgirdwood Done.